manager/integration/tests/test_basic.py

```diff
@@ -3658,3 +3660,128 @@ def test_volume_toomanysnapshots_condition(client, core_api, volume_name): # NOQ
 
     wait_for_volume_condition_toomanysnapshots(client, volume_name,
                                                "status", "False")
+
+
+def test_expand_lhv(client, core_api, volume_name, pod):  # NOQA
+    """
+    test expand lhv volume
+
+    1. Create LHV volume with size '1Gi'
+    2. attach and detach
+    3. Expand volume size to '2000000000/2G' and
+        check if size round up '2000683008'
+    4. attach and detach
+    5. Expand volume size to '2Gi' and check if size is '2147483648'
+    """
+
+    volume = create_and_check_volume(client, volume_name, 2, str(1 * Gi))
+
+    self_hostId = get_self_host_id()
+    volume.attach(hostId=self_hostId, disableFrontend=False)
+    wait_for_volume_healthy(client, volume_name)
+    volume.detach(hostId="")
+    volume = wait_for_volume_detached(client, volume_name)
+
+    volume.expand(size="2000000000")
+    wait_for_volume_expansion(client, volume_name)
+    volume = client.by_id_volume(volume_name)
+    assert volume.size == "2000683008"
+
+    self_hostId = get_self_host_id()
+    volume.attach(hostId=self_hostId, disableFrontend=False)
+    wait_for_volume_healthy(client, volume_name)
+    volume.detach(hostId="")
+    volume = wait_for_volume_detached(client, volume_name)
+
+    volume.expand(size=str(2 * Gi))
+    wait_for_volume_expansion(client, volume_name)
+    volume = client.by_id_volume(volume_name)
+    assert volume.size == "2147483648"
+
+    self_hostId = get_self_host_id()
+    volume.attach(hostId=self_hostId, disableFrontend=False)
+    wait_for_volume_healthy(client, volume_name)
+    volume.detach(hostId="")
+    volume = wait_for_volume_detached(client, volume_name)
+
+    client.delete(volume)
+    wait_for_volume_delete(client, volume_name)
+
+
+def test_expand_pvc(client, core_api, volume_name, pod):  # NOQA
+    """
+    test expand lhv volume
```

Comment:
expression `lhv volume` is weird. `lhv` is the alias of `longhorn volume`.

Reply:
changed